docs/connector-development/connector-specification-reference.md

```diff
@@ -3,46 +3,78 @@ The [connector specification](../understanding-airbyte/airbyte-specification.md#
 
 
 ### Using `oneOf`s 
-In some cases, a connector needs to accept one out of many options. For example, a connector might need to know the compression codec of the file it will read. In JSONSchema, this can be expressed using the [oneOf](https://json-schema.org/understanding-json-schema/reference/combining.html#oneof) keyword.
+In some cases, a connector needs to accept one out of many options. For example, a connector might need to know the compression codec of the file it will read, which will render in the Airbyte UI as a list of the available codecs. In JSONSchema, this can be expressed using the [oneOf](https://json-schema.org/understanding-json-schema/reference/combining.html#oneof) keyword.
+
+{% hint style="info" %}
+Some connectors may follow an older format for dropdown lists, we are currently migrating away from that to this standard.
+{% endhint %}
 
 In order for the Airbyte UI to correctly render a specification, however, a few extra rules must be followed: 
 
 1. The top-level item containing the `oneOf` must have `type: object`.
 2. Each item in the `oneOf` array must be a property with `type: object`.
-3. One `string` field with the same property name must be consistently present throughout each object inside the `oneOf` array. In each instance of this field, there must be a [`const`](https://json-schema.org/understanding-json-schema/reference/generic.html#constant-values) value unique to that `oneOf` option. For example: 
-    ```
-    {
-      "dropdown_list_option": {
+3. One `string` field with the same property name must be consistently present throughout each object inside the `oneOf` array.
+
+Let's look at the [source-file](../integrations/sources/file.md) implementation as an example. In this example, we have `provider` as a dropdown
+list option, which allows the user to select what provider their file is being hosted on. We note that the `oneOf` keyword lives under the `provider` object as follows:
+
+It is required to add a [`const`](https://json-schema.org/understanding-json-schema/reference/generic.html#constant-values) value unique to that `oneOf` option.
```

Comment:
can you add this line next to the sentence on line 16? the full requirement should ideally be one the same line, followed by the full example